Baikal/Utils/cl_program.cpp

```diff
@@ -262,6 +281,9 @@ CLWProgram CLProgram::GetCLWProgram(const std::string &opts)
             // Save binaries
             result.GetBinaries(0, binary);
             SaveBinaries(cached_program_path, binary);
+
+            // Block other workers until binary cache generated
+            cache_lock.unlock();
```

Comment:
This unlock is excessive since the cache_lock goes out of scope right after it.

Reply:
Removed